--- src/osgPlugins/ffmpeg/ReaderWriterFFmpeg.cpp.orig	2012-03-21 18:36:20.000000000 +0100
+++ src/osgPlugins/ffmpeg/ReaderWriterFFmpeg.cpp	2012-11-24 14:39:13.000000000 +0100
@@ -88,13 +88,16 @@
         {
             return readImageStream(filename, NULL);
         }
-
+#if LIBAVFORMAT_BUILD < (54<<16 | 29<<8 | 0)
         osg::ref_ptr<osgFFmpeg::FFmpegParameters> parameters(new osgFFmpeg::FFmpegParameters);
         parseOptions(parameters.get(), options);
         if (parameters->isFormatAvailable())
         {
             return readImageStream(filename, parameters.get());
         }
+#else
+        readImageStream(filename, 0);
+#endif
 
         if (! acceptsExtension(ext))
             return ReadResult::FILE_NOT_HANDLED;
@@ -105,8 +108,12 @@
 
         if (path.empty())
             return ReadResult::FILE_NOT_FOUND;
-
+#if LIBAVFORMAT_BUILD < (54<<16 | 29<<8 | 0)
         return readImageStream(path, parameters.get());
+#else
+        return readImageStream(path, 0);
+
+#endif
     }
 
     ReadResult readImageStream(const std::string& filename, osgFFmpeg::FFmpegParameters* parameters) const
@@ -132,7 +139,9 @@
                  itr != supportedOptList.end(); ++itr)
             {
                 const std::string& name = itr->first;
+#if LIBAVFORMAT_BUILD < (54<<16 | 29<<8 | 0)
                 parameters->parse(name, options->getPluginStringData(name));
+#endif
             }
         }
     }
