--- src/osgPlugins/ffmpeg/FFmpegHeaders.hpp.orig	2012-01-05 15:07:56.000000000 +0100
+++ src/osgPlugins/ffmpeg/FFmpegHeaders.hpp	2012-11-24 15:07:07.000000000 +0100
@@ -3,19 +3,27 @@
 #define HEADER_GUARD_FFMPEG_HEADERS_H
 
 
+//FFMPEG > 0.7 compatibility
+//when AVFormatParameter was deprecated for AVDictionary
+#define LIBAVFORMAT_BUILD_PARAM2DICT (53<<16 | 2<<8 | 0)
+//when AVERROR_... was deprecated for AVERROR(e)
+#define LIBAVUTIL_BUILD_AVERRORFCT (50<<16 | 12<<8 | 0) 
+//when CODEC_TYPE_... was deprecated for AVMEDIA_TYPE_...
+#define LIBAVUTIL_BUILD_CODEC2AVMEDIA (50<<16 | 14<<8 | 0) 
+
 extern "C"
 {
 #define __STDC_CONSTANT_MACROS
 #define FF_API_OLD_SAMPLE_FMT 0
 #include <errno.h>    // for error codes defined in avformat.h
 #include <stdint.h>
-#include <avcodec.h>
-#include <avformat.h>
-#include <avdevice.h>
-#include <mathematics.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libavdevice/avdevice.h>
+#include <libavutil/mathematics.h>
 
 #ifdef USE_SWSCALE    
-    #include <swscale.h>
+    #include <libswscale/swscale.h>
 #endif
 
 #if LIBAVUTIL_VERSION_INT <  AV_VERSION_INT(50,38,0)
